lib/split/helper.rb

```diff
@@ -95,23 +95,39 @@ def begin_experiment(experiment, alternative_name = nil)
 
     # alternative can be either a string or an array of strings
     def ensure_alternative_or_exclude(experiment_name, alternatives)
-      return if @alternative_ensured
       alternatives = Array(alternatives)
+      experiment   = ExperimentCatalog.find(experiment_name)
+
+      # This is called if the experiment is not yet created. Since we can't created (we need the alternatives
+      # configuration to be passed to this method), we exclude the user, and when `ab_test` is called after
+      # the experiment is created. This means the first user will be excluded from the test and will have
+      # the control alternative
+      if !experiment
+        experiment = Experiment.new(experiment_name)
+        exclude_visitor(experiment)
+        return
+      end
 
-      # if ab_user was not called before, then force the alternative and manually increment the counter.
-      unless experiment = ExperimentCatalog.find(experiment_name)
-        ab_user[experiment_name] = default_alternative = alternatives.first
-        Split::Alternative.new(default_alternative, experiment_name).increment_participation
-        @alternative_ensured = true
+      # If the user doesn't have an alternative set, we override the alternative and we manually increment the
```

Comment:
` If the user doesn't have an alternative set, we override the alternative ` It doesn't makes much sense